src/org/etools/j1939_84/bus/MultiQueue.java

```diff
@@ -34,8 +33,7 @@ synchronized public void add(T v) {
 
     @Override
     public void close() {
-        // close all of the spliterators.
-        spliterators.values().forEach(SpliteratorImplementation<T>::close);
+        spliterators.values().forEach(SpliteratorImplementation::close);
```

Comment:
@battjt , when the bus is closed, in theory we should be closing all the streams reading the bus:

```
spliterators.keySet().forEach(BaseStream::close);
```

However, doing that causes test failures.  Are the tests valid in this case?

Reply:
A stream can only be read once and once it is read, it is closed.  I don't think that we want to close the stream when the bus is closed.  The bus fills the stream; it doesn't really back the stream.  It would be like throwing out the pan of water when the faucet is turned off.  It makes sense to use the pan of water after the faucet is turned off.

That said, the implementation empties the stream on bus close (SpliteratorImplementation.close() empties the stream).  It only happens in the tests.  In the application, we don't stop and start the bus.

Closing the Stream on bus close causes tests to fail.  Those tests are not valid.  they are checking that that packets are not read after the bus is closed.